### DISCUSSION TITLE
Support for all builtin types?

### DISCUSSION THREAD

**Discussion by caniko** [pydantic/pydantic#4973] [Feature Request]
I was a little surprised in #4972 that builtin types are not supported natively by Pydantic, and wanted to understand why.

**Reasoning**
Builtin elements should support other _applicable_ builtin elements. This principle should extend to Pydantic because it fills such an important void in the Python ecosystem of builtins.

The void refers to the incompleteness of `dataclasses`.

**The bug: A logical bug**
I would at least not call builtins arbitrary as it is categorically wrong. Following the reasoning that Pydantic implicitely supports a subset of builtins, your selection of types to support are what becomes arbitrary. I just don't think builtin types are arbitrary, and I would like to understand why you think this.

**Suggestion**
All builtin types are at least type-checked, just like (not as) arbitrary 

**Feedback to @samuelcolvin**
It is harsh to expect me know that builtin types are not supported because their support is omitted from the docs. Please make a not supported section in the docs if this list of omitted builtin tools are long. This instance is too arbitrary to know ahead of making an Issue.

**Comment by samuelcolvin** [comment]
I'm happy to add support for more builtins, and if we can cover them all, to then guarantee to support all buitins.

**Reply by samuelcolvin** [reply]
@caniko maybe you could start by creating an issue with a list of all builtins, with:
* whether they're support now
* whether it makes sense to support them
* any extra notes
